src/ConcreteOperations/intersection.jl

```diff
@@ -89,6 +89,40 @@ function intersection(L1::Line{N}, L2::Line{N}
     end
 end
 
+"""
+    intersection(X::LineSegment{N}, hp::Line{N}
+                ) where {N<:Real}
+Compute the intersection of a line and a line segment.
+### Input
+- `a`  -- LineSegment
+- `b` -- Line
+### Output
+If the sets do not intersect, the result is the empty set.
+Otherwise the result is the singleton or line segment that describes the intersection.
+"""
+
+function intersection(a::LineSegment{N}, b::Line{N}) where {N<:Real}
+  # cast a as line
+  ap = Line(a.p, a.q)
+  # find intersection between a' and b
+  m = intersection(ap, b)
+  if m == b
+    # if this equals b, then all of the segment is the intersection
+    return a
```

Comment:
I would rather check for `m == ap` (or swap the arguments to `intersection`) because the implementation returns the first argument if they are equal (you could hence even check for `===`) and I am not sure what happens if they are only approximately equal.